sessionctx/stmtctx/stmtctx.go

```diff
@@ -486,7 +485,6 @@ func (sc *StatementContext) ResetForRetry() {
 	sc.TableIDs = sc.TableIDs[:0]
 	sc.IndexNames = sc.IndexNames[:0]
 	sc.TaskID = AllocateTaskID()
-	sc.CheckKeyExists = make(map[string]struct{})
```

Comment:
We need to reset check exists flags on the keys if the statement execution has failed and statement will be retried, like the pessimistic write conflict retry.

Reply:
`handlePessimisticLockError` will rollback all changes made in the current statement by `StmtRollback`. `KeyFlags` will be discarded in this step.